tests/unit/amo/reducers/test_collections.js

```diff
@@ -682,6 +677,33 @@ describe(__filename, () => {
 
       expect(state.editingCollectionDetails).toEqual(false);
     });
+
+    // See: https://github.com/mozilla/addons-frontend/issues/7412
+    it('resets the addonInCollections map when location changes', () => {
```

Comment:
This is the test case for this patch.

Other changes are `s/username/userId/`...